idmapping-rest/src/test/java/org/uniprot/api/idmapping/service/impl/PIRServiceImplTest.java

```diff
@@ -170,7 +167,39 @@ void testThrowsExceptionWhenDuplicateIds() {
                         "http://localhost", new HttpEntity<>(map, HTTP_HEADERS), String.class))
                 .thenReturn(ResponseEntity.ok().body("P00001\tP00001"));
 
-        Assertions.assertThrows(
-                IllegalStateException.class, () -> pirService.mapIds(request, "dummyJobId"));
+        IdMappingResult idMappingResult = pirService.mapIds(request, "dummyJobId");
+        assertThat(
+                idMappingResult.getMappedIds(),
+                contains(new IdMappingStringPair("P00001.1", "P00001")));
```

Comment:
I tested this use case in legacy and it returns without version.
In Legacy it would return "P00001", "P00001" --> it also would return that it found different versions, but I think it is another Jira.
Let me know if you have any doubt This is my test URL for legacy: https://legacy.uniprot.org/uniprot/?query=yourlist:M20230420A084FC58F6BBA219896F365D15F2EB4454E3D78&sort=yourlist:M20230420A084FC58F6BBA219896F365D15F2EB4454E3D78&columns=yourlist(M20230420A084FC58F6BBA219896F365D15F2EB4454E3D78),id,entry%20name,reviewed,protein%20names,genes,organism,length,comment(INVOLVEMENT%20IN_DISEASE)

Reply:
I think, legacy is not correct. We are returning correct mapping for version from beginning. 
if you map P05067.304 then we get P05067.304 -> P05067 (which is more consistent i would say)